modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+            switch (duration) {
+                case SECONDS:
+                    retentionPeriods.put(duration, sec);
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(duration, mins);
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, hours);
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, days);
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(duration, months);
+                    break;
+                case YEARS:
+                    retentionPeriods.put(duration, years);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        if (annotationTypes.keySet().contains(SiddhiConstants.NAMESPACE_PURGE)) {
+            if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                    getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE))) {
+                String enable = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE);
+                if (!(enable.equals("true") || enable.equals("false"))) {
+                    throw new SiddhiAppCreationException("Undefined value for enable: " + enable + "." +
+                            " Please use true or false");
+                } else {
+                    purgingEnabled = Boolean.parseBoolean(enable);
+                }
+            }
+            if (purgingEnabled) {
+                // If interval is defined, default value of 15 min will be replaced by user input value
+                if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.NAMESPACE_INTERVAL))) {
+                    String interval = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                            getElement(SiddhiConstants.NAMESPACE_INTERVAL);
+                    purgeExecutionInterval = timeToLong(interval);
+                }
+                List<Annotation> retentions = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getAnnotations(SiddhiConstants.NAMESPACE_RETENTION);
+                if (Objects.nonNull(retentions) && !retentions.isEmpty()) {
+                    Annotation retention = retentions.get(0);
+                    List<Element> elements = retention.getElements();
+                    for (Element element : elements) {
+                        switch (durationNormalization(element.getKey())) {
+                            case SECONDS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
```

Comment:
This if and else blocks can be extracted out above the switch case and store the value in a variable. And in the switch case we can just assign that value to the correct variable.

Reply:
changed the implementation